### PR TITLE
Feature/add remote logger; Silence useless info

### PR DESCRIPTION
**Pull request by VinceCui**
### Describe what this PR does / why we need it

Support remote logger, so sealer can upload its logs to another server.

### CODE REVIEW

File: cmd/sealer/cmd/root.go [hunk 1]
```diff
@@ -81,6 +89,13 @@ func initConfig() {
 	if rootOpt.debugModeOn {
 		logrus.SetLevel(logrus.DebugLevel)
 	}
+
+	if rootOpt.hideLogColor {
```
Comment (justadogistaken) [inline cmd/sealer/cmd/root.go:93]:
```
logrus.SetFormatter(&logrus.TextFormatter{
			DisableColors: rootOpt.hideLogColor,
})
```
Would this be better?
I'm ok with both.

File: pkg/remote-logger/remote-logger.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	}
+
+	return &RemoteLogHook{
+		TaskName: taskName,
+		URL:      reqURL.String(),
+	}, err
+}
+
+// #nosec
+func httpSend(url string, method string, body []byte) error {
```
Comment (justadogistaken) [inline pkg/remote-logger/remote-logger.go:47]: I'd like to start an request explicitly.
Like httpPost/httpGet.
But I'm also ok with your way for http request.

**Comment by justadogistaken** [comment]
@VinceCui I suggest we merge multiple commits into single commit.

**Comment by kakaZhou719** [comment]
currently we use the default logrus configuration，my idea about this is could we setup  a new directory named logger under pkg, and save things like , logger init, file config ,remote-hooks, and so on. @VinceCui ,please correct me if I am mistaken.

**Comment by kakaZhou719** [comment]
currently we use the default logrus configuration，my idea about this is could we setup  a new directory named logger under pkg, and save things like , logger init, file config ,remote-hooks, and so on. @VinceCui ,please correct me if I am mistaken.

File: cmd/sealer/cmd/root.go [hunk 3]
```diff
@@ -59,8 +63,12 @@ func init() {
 	cobra.OnInitialize(initConfig)
 	rootCmd.PersistentFlags().StringVar(&rootOpt.cfgFile, "config", "", "config file of sealer tool (default is $HOME/.sealer.json)")
 	rootCmd.PersistentFlags().BoolVarP(&rootOpt.debugModeOn, "debug", "d", false, "turn on debug mode")
+	rootCmd.PersistentFlags().BoolVar(&rootCmd.SilenceUsage, "silence-usage", false, "silence the usage when fail")
```
Comment (nashtsai) [inline cmd/sealer/cmd/root.go:66]: a common flag name should be `--quiet`, short being`-q`

File: cmd/sealer/cmd/root.go [hunk 4]
```diff
@@ -59,8 +63,12 @@ func init() {
...
 	rootCmd.PersistentFlags().BoolVar(&rootOpt.hideLogTime, "hide-time", false, "hide the log time")
 	rootCmd.PersistentFlags().BoolVar(&rootOpt.hideLogPath, "hide-path", false, "hide the log path")
+	rootCmd.PersistentFlags().BoolVar(&rootOpt.hideLogColor, "hide-color", false, "hide the log color")
```
Comment (nashtsai) [inline cmd/sealer/cmd/root.go:69]: I would suggest use `--color=[never, always, auto]` just like `grep` cmd

Comment (allencloud) [reply cmd/sealer/cmd/root.go:69]: Nice, more reasonable.

File: cmd/sealer/cmd/root.go [hunk 5]
```diff
@@ -81,6 +89,13 @@ func initConfig() {
...
 		logrus.SetLevel(logrus.DebugLevel)
 	}
+
+	if rootOpt.hideLogColor {
+		logrus.SetFormatter(&logrus.TextFormatter{
+			DisableColors: true,
+		})
+	}
+
 	/*
```
Comment (allencloud) [inline cmd/sealer/cmd/root.go:99]: Please also directly remove the commented code.

File: pkg/remote-logger/remote-logger.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RemoteLogHook to send logs via remote URL.
+type RemoteLogHook struct {
+	sync.RWMutex
```
Comment (allencloud) [inline pkg/remote-logger/remote-logger.go:28]: defined this, but not used?

**Comment by nashtsai** [comment]
> ### Describe what this PR does / why we need it
> Support remote logger, so sealer can upload its logs to another server.
> 
> ### Does this pull request fix one issue?
> ### Describe how you did it
> ### Describe how to verify it
> ### Special notes for reviews

Can you provide background context  why this feature is needed?

File: cmd/sealer/cmd/root.go (hunk 1 above)
Comment (VinceCui) [reply cmd/sealer/cmd/root.go:93]: u are right

File: pkg/remote-logger/remote-logger.go (hunk 6 above)
Comment (VinceCui) [reply pkg/remote-logger/remote-logger.go:28]: used in line 97

**Review by bxy4543 — APPROVED** [review]

**Pull request merged**